### PR TITLE
Rewriting Vue Frontend

### PR DESCRIPTION
**Pull request by akaeyuhi**
Rewtring old frontend logic using React library and Redux state manager(with toolkit). No significant changes, only rewriting and refactor.

### CODE REVIEW

**Review by ILabiak — DISMISSED** [review]

File: src/client/src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { Route, Routes } from 'react-router-dom';
+import { AppRoutes, ProtectedAppRoutes } from './features/AppRoutes';
+import Header from './components/Header';
+import Footer from './components/Footer';
+import ProtectedRoute from './features/ProtectedRoute';
+import { useSelector } from 'react-redux';
+
+function App() {
+  const user = useSelector(state => state.account.user) || null;
```
Comment (ILabiak) [inline src/client/src/App.jsx:10]: maybe use useEffect() to set user data

File: src/client/src/app/actions/account/addUserRecipe.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    try {
+      const token = getState()?.account.token;
+      const headers = {
+        'Content-Type': 'application/json',
+        'authorization': token,
+      };
+      const response = await fetch('/api/user/recipes', {
+        method: 'PUT',
+        headers,
+        body: { recipeId: JSON.stringify(recipe.id) },
```
Comment (ILabiak) [inline src/client/src/app/actions/account/addUserRecipe.js:15]: make a variable const body = JSON.stringify({ recipeId: recipe.id }); and then pass it to fetch method

File: src/client/src/features/ValidationError.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+export default class ValidationError {
+  constructor(type, description) {
+    this.type = type;
+    this.description = description;
+  }
+
+  static Validate(data) {
```
Comment (ILabiak) [inline src/client/src/features/ValidationError.js:7]: class method should start with lowercase letter

File: src/client/src/features/ValidationError.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    const emailRegex = /^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$/;
+    const validationErrors = [];
+    if (!emailRegex.test(data.email)) validationErrors.push(new ValidationError(
+      'email', 'Invalid email',
+    ));
+    if (data.password.length < 8) validationErrors.push(new ValidationError(
```
Comment (ILabiak) [inline src/client/src/features/ValidationError.js:13]: maybe add `    if (!data.password) {
      validationErrors.push(new ValidationError('password', 'Password is required'));`

**Review by ILabiak — APPROVED** [review]

File: src/client/src/App.jsx (hunk 1 above)
Comment (akaeyuhi) [reply src/client/src/App.jsx:10]: In Redux it's ok to use it like that

**Review by ILabiak — APPROVED** [review]
Approved!

**Pull request merged**